### ISSUE TITLE
Submarine doesn't get saved between levels in multiplayer

### ISSUE THREAD

**Issue by Rokvach** [Regalis11/Barotrauma#9172]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

This issue seems to only exist in multiplayer campaign

During community testing we noticed that all cabinets on Typhon 2 reset after each level transition.

The cabinets fully reset to the starting items, so if you optain any new items and store them to the cabinets those items are gone in the next round, also you can duplicate items by looting the items on the cabinet and keeping them in your inventory before starting a new round. 

### Reproduction steps

1. Start multiplayer campaign with Typhon 2
2. Loot some items from the cabinets or place new items in them
3. Transition to next level
4. Notice that the items you took from the cabinet are in the cabinet again and the items you placed are gone

### Bug prevalence

Happens every time I play

### Version

0.18.2.0 (Unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by Rokvach** [comment]
So far tested by taking and adding new items to cabinets and the cabinet looks as it should in the next round. I will test the rest of the points with a better time next week.

**Comment by Rokvach** [comment]
Once issue which might not be related to this is that if I lose a drone then switch submarine and then switch back to the submarine which shouldn't have a drone, now has a drone which isn't located in the docking port

1. Start campaign with Remora (I tested in multiplayer)
2. Undock from outpost
3. Undock remora drone and drive far away fromt he main sub
4. Dock back to outpost
5. Buy and switch to a new sub
6. Redock to outpost
7. Switch back to Remora
8. Undock from outpost and notice that the remora drone is back but not in the correct place

![image](https://user-images.githubusercontent.com/7299826/172312819-498516f7-df1c-4004-8e5e-403294e7ce2a.png)

**Comment by Regalis11** [comment]
Linked to an incorrect ticket in https://github.com/Regalis11/Barotrauma-development/commit/fba4ab583b83dfb5995aa9cc9afdb9737015f4d6, should've been https://github.com/Regalis11/Barotrauma/issues/9305

**Comment by Regalis11** [comment]
Hopefully fixed in https://github.com/Regalis11/Barotrauma-development/commit/d5897797fe0a072dcab6e98d0a6429e3c382939e. I also noticed another issue not related to submarine switching while debugging this:

> - Fixed linked sub spawning inside the main sub if you've loaded into a level in which it was left behind, and then dock back to the outpost (see https://github.com/Regalis11/Barotrauma/issues/9172).
> - Fixed linked sub spawning inside the main sub if you dock to an outpost when the linked sub is undocked, but not far enough from the main sub to be considered "left behind".

**Comment by Rokvach** [comment]
I didn't spot any further issues, working correctly. This ticket can be closed once it has passed review

**Comment by Rokvach** [comment]
Tested this again after recent changes and didn't notice any issues. Closing.
